newIDE/app/src/GameDashboard/LeaderboardAdmin/index.js

```diff
@@ -0,0 +1,820 @@
+import SelectField from '../../UI/SelectField';
+import SelectOption from '../../UI/SelectOption';
+
+const breakUuid = (uuid: string): string => `${uuid.split('-')[0]}-...`;
+
+type Props = {| onLoading: boolean => void |};
+type ContainerProps = {| ...Props, gameId: string |};
+
+type ApiError = {|
+  action:
+    | 'entriesFetching'
+    | 'entryDeletion'
+    | 'leaderboardsFetching'
+    | 'leaderboardNameUpdate'
+    | 'leaderboardSortUpdate'
+    | 'leaderboardPlayerUnicityDisplayChoiceUpdate'
+    | 'leaderboardCreation'
+    | 'leaderboardReset'
+    | 'leaderboardDeletion',
+  message: React.Node,
+  itemId?: string,
+|};
+
+const CenteredError = ({ children }: {| children: React.Node |}) => (
+  <Column expand justifyContent="center" alignItems="center">
+    <Line>{children}</Line>
+  </Column>
+);
+
+const styles = {
+  leftColumn: { display: 'flex', flexDirection: 'column', flex: 1, padding: 5 },
+  rightColumn: {
+    display: 'flex',
+    flexDirection: 'column',
+    flex: 2,
+  },
+};
+
+const LeaderboardAdmin = ({ onLoading }: Props) => {
+  const isOnline = useOnlineStatus();
+  const windowWidth = useResponsiveWindowWidth();
+  const [isEditingName, setIsEditingName] = React.useState<boolean>(false);
+  const [isRequestPending, setIsRequestPending] = React.useState<boolean>(
+    false
+  );
+  const [newName, setNewName] = React.useState<string>('');
+  const [newNameError, setNewNameError] = React.useState<?string>(null);
+  const newNameTextFieldRef = React.useRef<?TextField>(null);
+  const [apiError, setApiError] = React.useState<?ApiError>(null);
+
+  const {
+    leaderboards,
+    listLeaderboards,
+    currentLeaderboard,
+    createLeaderboard,
+    selectLeaderboard,
+    updateLeaderboard,
+    resetLeaderboard,
+    deleteLeaderboard,
+    deleteLeaderboardEntry,
+    displayOnlyBestEntry,
+    setDisplayOnlyBestEntry,
+    fetchLeaderboardEntries,
+    browsing: { entries, goToNextPage, goToPreviousPage, goToFirstPage },
+  } = React.useContext(LeaderboardContext);
+
+  const disableActions = React.useCallback(
+    (yesOrNo: boolean) => {
+      setIsRequestPending(yesOrNo);
+      onLoading(yesOrNo);
+    },
+    [onLoading]
+  );
+
+  const _updateLeaderboard = async (
+    i18n: I18nType,
+    payload: {|
+      name?: string,
+      sort?: LeaderboardSortOption,
+      playerUnicityDisplayChoice?: LeaderboardPlayerUnicityDisplayOption,
+    |}
+  ) => {
+    setNewNameError(null);
+    if (payload.name !== undefined && payload.name.length === 0) {
+      setNewNameError(
+        i18n._(
+          t`Please enter a name that is at least one character long and 50 at most.`
+        )
+      );
+      return;
+    }
+    disableActions(true);
+    setApiError(null);
+    try {
+      await updateLeaderboard(payload);
+      if (payload.name) setIsEditingName(false);
+    } catch (err) {
+      console.error(`An error occurred when updating leaderboard: ${err}`);
```

Comment:
For errors in the Browser console, prefer to pass the error as a separate object: console.error("An error [...]", err);

so that it's not stringified and keep containing all the useful info.